lib/src/wiggle_abi/shielding.rs

```diff
@@ -7,16 +7,29 @@ impl fastly_shielding::FastlyShielding for Session {
         &mut self,
         memory: &mut wiggle::GuestMemory<'_>,
         name: wiggle::GuestPtr<str>,
-        _out_buffer: wiggle::GuestPtr<u8>,
+        out_buffer: wiggle::GuestPtr<u8>,
         _out_buffer_max_len: u32,
     ) -> Result<u32, Error> {
         // Validate the input name and then return the unsupported error.
         let name_bytes = memory.to_vec(name.as_bytes())?;
-        let _name = String::from_utf8(name_bytes).map_err(|_| Error::InvalidArgument)?;
+        let name = String::from_utf8(name_bytes).map_err(|_| Error::InvalidArgument)?;
+
+        let mut shield_config = name.split('-');
+
+        let first = shield_config.next().expect("first is required");
```

Comment:
I leave it up to your discretion to rename these as makes sense, I'm not 100% sure what the blocks refer to.